src/mds/MDCache.cc

```diff
@@ -13668,44 +13672,63 @@ void MDCache::dispatch_quiesce_inode(const MDRequestRef& mdr)
   }
 
   if (!(mdr->locking_state & MutationImpl::ALL_LOCKED)) {
-    MutationImpl::LockOpVec lov;
+    {
+      /* knowing that quiescelock is always locked first... */
+      MutationImpl::LockOpVec lov;
+      lov.add_xlock(&in->quiescelock); /* !! */
+      if (!mds->locker->acquire_locks(mdr, lov, nullptr, false, true)) {
+        return;
+      }
+    }
 
-    lov.add_xlock(&in->quiescelock); /* !! */
-    lov.add_rdlock(&in->policylock); /* for F_QUIESCE_BLOCK test */
+    /* kick cap issuance without waiting for cap-related locks */
```

Comment:
I don't see a reason to break the quiesce and the policy lock into two calls, let's acquire them together. This will be the first step to improve our handling of the quiesce.block (along with renaming it to quiesce. skip) flag to avoid recalling caps for files which were marked to skip the quiesce.

Reply:
The usual concern is lock order dependency deadlocks but in this case we have

```
#define CEPH_LOCK_IQUIESCE    (1 << 4)  /* mds internal */
#define CEPH_LOCK_ISNAP       (1 << 5)  /* snapshot lock. MDS internal */
#define CEPH_LOCK_IPOLICY     (1 << 6)  /* policy lock on dirs. MDS internal */
```

and we don't acquire the snaplock. I'll make this change and add a note.